src/test/test-str-slice.c

```diff
@@ -108,6 +108,80 @@ test_str_slice_str_eq(void)
 #undef TEST_NOT_EQUAL
 }
 
+DEFINE_TEST(test_str_slice_remove_leading_whitespace);
+
+static void
+test_str_slice_remove_leading_whitespace(void)
+{
+    unsigned int i;
+
+#define TEST_EQUAL(X) { SOL_STR_SLICE_LITERAL(X), true }
+#define TEST_NOT_EQUAL(X) { SOL_STR_SLICE_LITERAL(X), false }
+
+    static const struct {
+        struct sol_str_slice input;
+        bool equal;
+    } table[] = {
+        TEST_NOT_EQUAL(" with one leading whitespace"),
+        TEST_NOT_EQUAL("  with two leading whitespace"),
+        TEST_NOT_EQUAL(" "),
+        TEST_NOT_EQUAL("\twith one leading whitespace"),
+        TEST_NOT_EQUAL("\t\twith two leading whitespace"),
+        TEST_NOT_EQUAL("\t"),
+        TEST_NOT_EQUAL("\nwith one leading whitespace"),
+        TEST_NOT_EQUAL("\n\nwith two leading whitespace"),
+        TEST_NOT_EQUAL("\n"),
+        TEST_EQUAL(""),
+        TEST_EQUAL("without leading whitespace"),
```

Comment:
Ok,  I thought I had added all the cases you've asked...  Will add these ones as well,  thanks.